### ISSUE TITLE
TensorBoard Embedding Projector Metadata filepath format not clear

### ISSUE THREAD

**Issue by suiluj** [tensorflow/tensorflow#47565]
## URL(s) with the issue:

- https://www.tensorflow.org/tensorboard/tensorboard_projector_plugin
- https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard
  - description of argument *embeddings_metadata*; See details link is gone: https://www.tensorflow.org/how_tos/embedding_viz/#metadata_optional

## Description of issue (what needs changing):

At the moment i try to use the embedding projector when using multiple runs (with a subfolder named by date for every training run)

I am not able to get the embedding projection including the metadata.tsv file for description labels (instead of just showing the index).

Everything does work as expected when i follow this tutorial for only one run: https://www.tensorflow.org/tensorboard/tensorboard_projector_plugin

The embedding projector does show my custom labels when hovering over points.

But the embedding projector was not able to read the metadata.tsv file when i saved multiple runs in different subfolders.
I tried a lot of combinations (storing the metadata.tsv file in the root logs folder or in the subfolder of the specific run and so on)
It does not recognise the metadata file. So when hovering over points it only shows the index.

At first i used the manual process described in the tensorboard_projector_plugin tutorial.

Later i tried to use the TensorBoard callback with the arguments *embeddings_freq=1, embeddings_metadata=(tried different parameters as dict or string)*

I would be very happy to get clear instruction on how to define the metadata file path in the current version of tensorflow/tensorboard. It seems that a few other people have the same problem. Perhaps it did change with a newer version.

I am not sure how to reference the path:
- relative path from subfolder of run
- relative path from tensorflow log folder root / (experiment root in case of multiple subfolders)
- use backslash or forward slash (does it make a difference when using windows vs. unix)
- how to reference the different layers when using the dict format in the TensorBoard callback instead of a string (in case of different embedding metadata for different embedding layers
  - just use layer name as string which ist defined in model?
  - or use some string with the suffix */.ATTRIBUTES/VARIABLE_VALUE* (as described in the projector tutorial)?

I really tried a lot of things so i would be very happy to finally get a solution or clear description.

Thank you.

Reference links to issues with the same problem:

- https://github.com/tensorflow/tensorflow/issues/33967
- https://stackoverflow.com/questions/41708106/linking-tensorboard-embedding-metadata-to-checkpoint
- https://stackoverflow.com/questions/42679552/how-can-i-select-which-checkpoint-to-view-in-tensorboards-embeddings-tab

**Comment by jvishnuvardhan** [comment]
@suiluj I think this is more related to Tensorboard (TB) repo where experts will resolve issues related to TB. [Here](https://github.com/tensorflow/tensorboard/issues) is the repo.

AFAIK, you can create multiple log files and load each one as shown in that example. I tried three different training runs and collected logs files as shown below. I was able to open each log to visualize it.

<img width="324" alt="Screen Shot 2021-03-05 at 3 40 21 PM" src="https://user-images.githubusercontent.com/46058173/110186362-d6c40080-7dc9-11eb-85c8-d02fcdb4efe0.png">
 
My question is do you want to load all three (or many) logs and compare them?
[Here](https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/tensorboard_projector_plugin.ipynb#scrollTo=PtL_KzYMBIzP) is a gist for reference. Thanks! 

Please post this in TB repo and close here. Thanks!

**Comment by suiluj** [comment]
hello @jvishnuvardhan yes i know how to train multiple runs and log everything in subfolders. Event the embeddings are visible.
The main point is that i was not able to show the visualizations including my metadata.

I found an older issue which had kind of the same problem with the projector_config which includes the metadata path.

Should i create a completely new issue in the tensorboard repo either way or is it enough to link these issues like i did?

**Comment by jvishnuvardhan** [comment]
@suiluj Please create a completely new issue in the Tboard repo. When you create a new issue it can be assigned to an expert who can resolve the issue faster. 

Please close the issue here after creating issue in Tboard repo. Thanks!

**Comment by jvishnuvardhan** [comment]
@suiluj I am closing this issue here as you are going to open it in TB repo. Thanks!
